targets/TARGET_Cypress/TARGET_PSOC6/TARGET_CY8CPROTO_064_SB/device/COMPONENT_CM4/TOOLCHAIN_ARM/cyb06xx7_cm4_dual.sct

```diff
@@ -168,15 +172,26 @@ LR_IROM1 FLASH_START FLASH_SIZE
     }
 
     ; Application heap area (HEAP)
-    ARM_LIB_HEAP  +0 EMPTY RAM_START+RAM_SIZE-STACK_SIZE-ImageLimit(RW_IRAM1)
+    ARM_LIB_HEAP  +0 ALIGN 4 EMPTY RAM_START+RAM_SIZE-MBED_BOOT_STACK_SIZE-MBED_DATA_SHARED_SIZE-ImageLimit(RW_IRAM1)
     { 
     }
-    
+
     ; Stack region growing down
-    ARM_LIB_STACK RAM_START+RAM_SIZE EMPTY -STACK_SIZE 
+    ARM_LIB_STACK RAM_START+RAM_SIZE-MBED_DATA_SHARED_SIZE ALIGN 4 EMPTY -MBED_BOOT_STACK_SIZE 
     {
     }
 
+    ; region RAM overflowed with Stack area
+    ScatterAssert(ImageBase(ARM_LIB_STACK) + ImageLength(ARM_LIB_STACK) == RAM_START+RAM_SIZE-MBED_DATA_SHARED_SIZE)
+
+    ; Shared region growing down
+    ARM_LIB_SHARED RAM_START+RAM_SIZE ALIGN 4 EMPTY -MBED_DATA_SHARED_SIZE 
```

Comment:
Why is shared region growing down? The definition matches neither the Mbed OS GCC linker script nor the TF-M scatter file (https://git.trustedfirmware.org/trusted-firmware-m.git/tree/platform/ext/target/cypress/psoc64_1m/Device/Source/armclang/psoc6_ns.sct?h=feature-twincpu#n48)

Reply:
I don't understand this comment. The shared region is last region defined on RAM.